staticcheck/sa2003/sa2003.go

```diff
@@ -21,6 +21,7 @@ var SCAnalyzer = lint.InitializeAnalyzer(&lint.Analyzer{
 	},
 	Doc: &lint.RawDocumentation{
 		Title:    `Deferred \'Lock\' right after locking, likely meant to defer \'Unlock\' instead`,
+		Text:     `Deferring \'Lock\' immediately after locking will cause a deadlock. Use \'Unlock\' instead.`,
```

Comment:
It doesn't necessarily cause a deadlock (and there are indeed possible false positives with this check if you write hard to follow code). 

At the end of the day, this check just flags a probable typo, I'm not sure what we could document about that.